pulpcore/app/models/publication.py

```
@@ -69,6 +70,29 @@ def create(cls, repository_version, pass_through=False):
             Adds a Task.created_resource for the publication.
         """
         with transaction.atomic():
+            # Check if at least some remote artifacts have allowed checksum
+            for ca in ContentArtifact.objects.filter(
+                content__in=repository_version.content, artifact=None
+            ):
+                allowed_checksum_content = False
+                remote_artifacts = ca.remoteartifact_set.all()
+                for checksum in Artifact.DIGEST_FIELDS:
+                    if any([ra for ra in remote_artifacts if getattr(ra, checksum, None)]):
+                        allowed_checksum_content = True
+                        break
+                # We don't need to continue, allowed checksum type found
+                if allowed_checksum_content:
+                    break
+            else:
+                # In case of no remote artifact to check, ensure exception won't be raised
```

Given that we are looping through content artifacts with no artifacts, this would mean that the content artifact has no artifact or remote artifacts. This case should probably raise an exception. I'd probably just raise the ValueError you have below by removing these lines.